src/Toolkit/Str.php

```diff
@@ -264,17 +266,33 @@ public static function contains(string $string = null, string $needle, bool $cas
      * according to locale settings
      *
      * @param int|null $time
-     * @param string|null $format
-     * @param string $handler date or strftime
+     * @param string|array|\IntlDateFormatter|null $format
+     * @param string $handler date, intl or strftime
      * @return string|int
      */
-    public static function date(?int $time = null, ?string $format = null, string $handler = 'date')
+    public static function date(?int $time = null, $format = null, string $handler = 'date')
     {
         if (is_null($format) === true) {
             return $time;
         }
 
-        // separately handle strftime to be able
+        // $format is an IntlDateFormatter instance
+        if (is_a($format, 'IntlDateFormatter') === true) {
+            return $format->format($time);
```

Comment:
`IntlDateFormatter::format()` doesn't accept `null` as the `$time`. So we need this:

```suggestion
            return $format->format($time ?? time());
```